File: client/Whisper/app/src/main/java/com/syang/whisper/security/AES.java
package com.syang.whisper.security;

import android.util.Base64;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

public class AES {
    private static final String ALGORITHM = "AES";
    private static final String PADDING = "AES/CBC/PKCS5Padding";

    public static String Base64Encrypt(String data, String key, String iv) {
        return Base64.encodeToString(encrypt(data, key, iv), Base64.DEFAULT);
    }

    public static String Base64Decrypt(String data, String key, String iv) {
        return decrypt(Base64.decode(data, Base64.DEFAULT), key, iv);
    }

    public static byte[] encrypt(String data, String key, String iv) {
        if (key == null) {
            key = "NOT_HACK_THE_KEY";
        }
        if (iv == null) {
            iv = "IT_IS_DEFAULT_IV";
        }
        IvParameterSpec mIV = new IvParameterSpec(iv.getBytes());
        SecretKeySpec mSecretKeySpec = new SecretKeySpec(key.getBytes(), ALGORITHM);
        try {
            Cipher cipher = Cipher.getInstance(PADDING);
            cipher.init(Cipher.ENCRYPT_MODE, mSecretKeySpec, mIV);
            return cipher.doFinal(data.getBytes());
        } catch (Exception ex) {
            return null;
        }
    }

    public static String decrypt(byte[] data, String key, String iv) {
        if (key == null) {
            key = "NOT_HACK_THE_KEY";
        }
        if (iv == null) {
            iv = "IT_IS_DEFAULT_IV";
        }
        IvParameterSpec mIV = new IvParameterSpec(iv.getBytes());
        SecretKeySpec mSecretKeySpec = new SecretKeySpec(key.getBytes(), ALGORITHM);
        try {
            Cipher cipher = Cipher.getInstance(PADDING);
            cipher.init(Cipher.DECRYPT_MODE, mSecretKeySpec, mIV);
            return new String(cipher.doFinal(data));
        } catch (Exception ex) {
            return "";
        }
    }
}

File: server/crypto/aes.js
const crypto = require('crypto');

const DEFAULT_IV = 'IT_IS_DEFAULT_IV';

const DEFAULT_KEY = 'NOT_HACK_THE_KEY';

module.exports = {
    encrypt: (data, key, iv) => {
        iv = iv || DEFAULT_IV;
        key = key || DEFAULT_KEY;
        const cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
        const encrypted = cipher.update(data, 'utf8', 'base64') + cipher.final('base64');
        return encrypted;
    },
    decrypt: (data, key, iv) => {
        iv = iv || DEFAULT_IV;
        key = key || DEFAULT_KEY;
        const decipher = crypto.createDecipheriv('aes-128-cbc', key, iv);
        const decrypted = decipher.update(data, 'base64', 'utf8') + decipher.final('utf8');
        return decrypted;
    }
}